### PR TITLE
Add llvm-mos trunk.

### PR DESCRIPTION
**Pull request by mysterymath**
This fetches the llvm-mos 6502 toolchain and SDK.

See http://llvm-mos.org and https://github.com/llvm-mos/llvm-mos-sdk for details.

### CODE REVIEW

File: bin/yaml/c.yaml [hunk 1]
```diff
@@ -28,6 +28,15 @@ compilers:
         - "2.17"
         - "2.18"
         - "2.19"
+    llvmmos:
+      type: tarballs
+      compression: xz
+      dir: llvm-mos-{name}
+      check_exe: bin/clang --version
+      url: https://github.com/llvm-mos/llvm-mos-sdk/releases/latest/download/llvm-mos-linux.tar.xz
```
Comment (mattgodbolt) [inline bin/yaml/c.yaml:36]: Can we have a tagged version for this? "trunk" only makes sense if we build/install this nightly (which this change won't do).

We can either:
- specify a named version here (well, as `{name}` here, and then with an appropraite `target:` below
- OR move this to the nightly section, and use `always install` type semantics

For the former obviously we'll need to keep adding here as new ones turn up and also keep updating the main CE repo. And it seems like this is fast moving and unstable, so probably not that.

For the second option you can add:
```yaml
if: nightly
type: nightlytarballs
```
here instead (well, not add the `type`, change the `type`. Hopefully that's clear.

**Review by mattgodbolt — COMMENTED** [review]
Thanks! A comment on this; it seems like this is a "trunk" build and there's two ways forward! Thank you for this contirbution, let me know what you think and if you want to change the PR please do, else let me know if you want us to make the change.

File: bin/yaml/c.yaml (hunk 1 above)
Comment (mysterymath) [reply bin/yaml/c.yaml:36]: Ah, thanks, I was wondering what exactly nightlytarballs did.

That seems like the way to go, at least until development on the compiler cools down somewhat. The compiler's already very usable, but it's still moving a bit too fast to cut a SemVer 1.0. When that time comes, that seems like it'd  also be a good time to switch to a versioned release here.

**Review by mattgodbolt — APPROVED** [review]

**Comment by mattgodbolt** [comment]
Installed just fine:
```
Installing compilers/c/llvmmos trunk
2022-11-15 03:43:54,873 lib.installation_context INFO     Fetching https://github.com/llvm-mos/llvm-mos-sdk/releases/latest/download/llvm-mos-linux.tar.xz (221417760 bytes)
2022-11-15 03:43:57,304 lib.installation_context INFO     100% of https://github.com/llvm-mos/llvm-mos-sdk/releases/latest/download/llvm-mos-linux.tar.xz
2022-11-15 03:43:57,306 lib.installation_context INFO     Piping to tar Jxf -
2022-11-15 03:44:42,661 lib.installation_context INFO     Moving from staging (/opt/compiler-explorer/staging/0fef56b7-fe0a-4bd4-8353-0ded80357a9c/llvm-mos) to final destination (/opt/compiler-explorer/llvm-mos-trunk-20221115)
2022-11-15 03:44:44,649 lib.installation_context INFO     Symlinking llvm-mos-trunk-20221115 to /opt/compiler-explorer/llvm-mos-trunk
2022-11-15 03:44:45,424 lib.ce_install  INFO     compilers/c/llvmmos trunk installed OK
1 packages installed OK, 0 skipped, and 0 failed installation
```

Runs ok too:

```
$ /opt/compiler-explorer/llvm-mos-trunk/bin/clang --version
clang version 16.0.0 (https://www.github.com/llvm-mos/llvm-mos.git 34d2de256c73a0c3f6d62c6bc67f32db85e77c1f)
Target: mos-unknown-unknown
Thread model: posix
InstalledDir: /opt/compiler-explorer/llvm-mos-trunk/bin
```

**Pull request merged**